apps/server/src/modules/tool/uc/lti-tool.uc.ts

```diff
@@ -0,0 +1,59 @@
+import { LtiToolRepo } from '@shared/repo';
+import { LtiToolDO } from '@shared/domain/domainobject/ltitool.do';
+import { ICurrentUser, IFindOptions, Page, Permission, User } from '@shared/domain';
+import { AuthorizationService } from '@src/modules/authorization';
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class LtiToolUc {
+	constructor(private readonly ltiToolRepo: LtiToolRepo, private readonly authorizationService: AuthorizationService) {}
+
+	async findLtiTool(
+		currentUser: ICurrentUser,
+		query: Partial<LtiToolDO>,
+		options: IFindOptions<LtiToolDO>
+	): Promise<Page<LtiToolDO>> {
+		const user: User = await this.authorizationService.getUserWithPermissions(currentUser.userId);
+		this.authorizationService.checkAllPermissions(user, [Permission.TOOL_VIEW]);
+
+		const tool: Promise<Page<LtiToolDO>> = this.ltiToolRepo.find(query, options);
+		return tool;
+	}
+
+	async getLtiTool(currentUser: ICurrentUser, toolId: string): Promise<LtiToolDO> {
+		const user: User = await this.authorizationService.getUserWithPermissions(currentUser.userId);
+		this.authorizationService.checkAllPermissions(user, [Permission.TOOL_VIEW]);
```

Comment:
there needs to be some kind of further check here, to prevent access to tools from foreign courses, or to those that are specifically hidden by the superhero.
Looking at the client, the only usecase for the get route seems to be for using a tool within a course, so I would suggest making this usecase specific to that.

(it probably makes sense to change the route too, something like courses/:id/ltitool/:id)